src/bio/bio_config.c

```diff
@@ -615,11 +624,12 @@ check_vmd_status(struct json_config_ctx *ctx, struct spdk_json_val *vmd_ss, bool
 {
 	int	rc;
 
+	D_ASSERT(vmd_enabled != NULL);
+	D_ASSERT(*vmd_enabled == false);
```

Comment:
Does this need to be an assertion? If the calling code changes somehow so that this was set to true elsewhere, should this really fail, or should it just return early?

Reply:
I think this is a valid assertion, it ensures that when check_vmd_status is called, vmd has not been enabled. check_vmd_status should only be called once and when it is vmd should not have been enabled yet, any situation to the contrary means the flow has been corrupted. if the flow is adjusted in the calling code then this will fail and prompt the person making the change to fix the flow.